### PR TITLE
Why Node.js is popular among developers

### PR DESCRIPTION
**Pull request by geoffrey45**
Created an article about the reasons behind Node.js popularity since its release.

I have covered what Node.js is, a little history about Node.js, Statistics proving its popularity, reasons behind its popularity, advantages of using Node in your projects and some application type it can be used to build.
I have included a hero image from unsplash, and 2 images showing survey statistics.

### CODE REVIEW

**Comment by hectorkambow** [comment]
Good afternoon Mani - wanted to ask if you could give this Node Js article a tech review whenever you have a moment. 
Before we publish it.
Thank you 🚀

File: articles/why-node-js-is-popular/index.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+The definition of Node.js is quite complex. It's not a programming language nor is it a framework or a library. On their official [website](https://nodejs.org/en), Node.js is defined as a JavaScript runtime built on Chrome's V8 JavaScript engine. Node.js can be simply referred to as a technology. It is mainly used for backend operations and is/can be used together with frontend JavaScript frameworks like [Angular](https://angularjs.org/), [React](https://reactjs.org/), [Vue](https://vuejs.org/) and others. Node.js is similar in design to Ruby's [Event Machine](https://github.com/eventmachine/eventmachine) and Python's [Twisted](https://twistedmatrix.com/trac/).
+
+### How and why did Node.js gain popularity?
+Node.js was introduced into the developer world in 2009 by the initial writer Ryan Dahl. Back in 2009, Apache HTTP server was a popular web server. But, it had a major problem, it could not handle a lot of concurrent connections. When Node.js was created it was able to solve the Apache problem. It offered scalability and parallel code execution. It introduced a new logic and approach for JavaScript server-side development. This has helped many developers.
```
Comment (manibatra) [inline articles/why-node-js-is-popular/index.md:23]: `the Apache problem` ? Possibly a little vague. Reading the next line one can infer it to be scalability and parallel code execution but if we are using it as a Noun it implies implicit knowledge. `the problems Apache had` might be better. 

`It introduced a new logic and approach for JavaScript server-side development.` this is vague too, some words on what new logic/approach was introduced will be better.

File: articles/why-node-js-is-popular/index.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+![most wanted technology 2020](/engineering-education/why-node-js-is-popular/stackoverflow2.jpg)
+[Image Source](https://insights.stackoverflow.com/survey/2020#technology-other-frameworks-libraries-and-tools-all-respondents3)
+
+According to Node.js [metrics](https://nodejs.org/metrics/), Node.js has been downloaded more than 1 billion times globally.
+
+### Why Node.js is popular among developers.
+
+#### 1. It is easy to learn Node.js.
+This is one of the major contributors to its wide usage.
+Among many developers, JavaScript skills are mandatory. Node.js uses JavaScript. So, learning Node.js is relatively easy and few weeks of learning can get you up and running. Without JavaScript Knowledge, learning Node.js will take a bit longer. Therefore, learning Node.js is not that tricky.
```
Comment (manibatra) [inline articles/why-node-js-is-popular/index.md:43]: `common` instead of `mandatory`?

File: articles/why-node-js-is-popular/index.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+#### 2. The scalability offered.
+Node.js is designed with scalable network applications in mind. Node.js can handle many concurrent requests. This is the main reason it quickly became popular among developers and large companies. With the ability to handle many simultaneous requests, the server does not strain. A callback is fired only after a request, but if there are no more jobs to be done, Node.js will go to sleep. Node.js is only online when needed. This avoids unnecessary usage of your RAM.
```
Comment (manibatra) [inline articles/why-node-js-is-popular/index.md:46]: `With the ability to handle many simultaneous requests, the server does not strain.`

`Can handle many simultaneous requests without straining the server` sounds better.

File: articles/why-node-js-is-popular/index.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+### 3. The Mobile-friendly, Cross-platform, and Dev-Friendly nature of Node.js.
+The ability to build flexible apps that run smoothly on any platform, be it Linux, Windows, or Mac OS gives Node.js a big thumbs up. There has been a rise in cross-platform app development in the last few years. The fear of apps not running on other operating systems is gone, as Node.js helped take care of that. It has enabled developers to write apps that run on different platforms with little or no modifications being made.
+
+#### 4. Node.js is light and fast.
+Node.js uses Google's V8 JavaScript engine which is built in C++. V8 engine handles memory allocation, compiles, and executes JavaScript with lightning speed. More about Google's V8 JS engine can be found in the V8 [docs](https://v8.dev/docs). Node.js uses a non-blocking model in performing operations. Node.js has published a full article on blocking and non-blocking models [here](https://nodejs.org/en/docs/guides/blocking-vs-non-blocking/). It handles requests made in a single asynchronous thread. This reduces CPU workloads as well as memory. This makes your app lightweight.
```
Comment (manibatra) [inline articles/why-node-js-is-popular/index.md:52]: `More details about Google's`

File: articles/why-node-js-is-popular/index.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+#### 5.The many hosting providers available.
+As a result of its growth in recent years, Many cloud-based hosting providers and web servers have added quite simple ways of hosting Node.js code. Some of these hosting providers include Heroku, Amazon Web Services, and DigitalOcean among others. There are also many detailed and simple guides on this.
+
+#### 6. Highly extensible
+Node.js does not provide everything that you need all at once. Instead, you customize or add what you need over time. This prevents having unnecessary libraries. [Npm](https://www.npmjs.com/) - **Node package manager** is the default package manager in Node.js. Here, you can browse and install over 1 million open source packages. You can also use it to identify and install project dependencies.
+
+#### 7. Its caching ability
+Node.js provides the ability to cache single modules. Caching is the storage of data for future requests. When a request is made, resources are fetched from the cache instead of the main servers. Thus, the web pages load faster and responds quickly.
```
Comment (manibatra) [inline articles/why-node-js-is-popular/index.md:61]: The sentence starts with caching of modules and then moves on to caching of pages? I am assuming we want to talk about caching modules here?

File: articles/why-node-js-is-popular/index.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+- Many more.
+
+### The advantages of using Node.js.
+ - It is fast and scales on-demand.
+ - Node.js modules and tools are easily available via [Npm](https://www.npmjs.com/).
+ - Node.js developers are easily available.
+ - Strong community and bug tracking team.
+ - It can be used to build a wide range of apps.
+ - It's Cross-platform and mobile-friendly.
+ - Hosting Node.js code is not a headache due to many supported hosting grounds.
```
Comment (manibatra) [inline articles/why-node-js-is-popular/index.md:84]: hosting platforms instead of grounds.

File: articles/why-node-js-is-popular/index.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+ - It's lightweight and fast.
+ - It's ability to handle many simultaneous requests.
+ - Fast code execution.
```
Comment (manibatra) [inline articles/why-node-js-is-popular/index.md:87]: covered in first point : ` It is fast and scales on-demand`

File: articles/why-node-js-is-popular/index.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+### The advantages of using Node.js.
+ - It is fast and scales on-demand.
+ - Node.js modules and tools are easily available via [Npm](https://www.npmjs.com/).
+ - Node.js developers are easily available.
+ - Strong community and bug tracking team.
+ - It can be used to build a wide range of apps.
+ - It's Cross-platform and mobile-friendly.
+ - Hosting Node.js code is not a headache due to many supported hosting grounds.
+ - It's lightweight and fast.
```
Comment (manibatra) [inline articles/why-node-js-is-popular/index.md:85]: `fast` : covered in first point ` It is fast and scales on-demand`

**Comment by geoffrey45** [comment]
I have committed the changes https://github.com/section-io/engineering-education/pull/233/commits/61a408bc1c34e3dadaf937bcad5a6ad993b8ff2e

**Pull request merged**